src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/validappeal/SyaAppealValidator.java

```diff
@@ -0,0 +1,163 @@
+import static uk.gov.hmcts.reform.sscs.ccd.callback.ValidationType.SYA_APPEAL;
+import static uk.gov.hmcts.reform.sscs.ccd.validation.sscscasedata.AddressValidator.IS_NOT_A_VALID_POSTCODE;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appeal;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseLink;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseLinkDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.service.CcdService;
+import uk.gov.hmcts.reform.sscs.ccd.validation.sscscasedata.AddressValidator;
+import uk.gov.hmcts.reform.sscs.ccd.validation.sscscasedata.AppealValidator;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+import uk.gov.hmcts.reform.sscs.service.DwpAddressLookupService;
+
+@Component
+@Slf4j
+public class SyaAppealValidator extends AppealValidator {
+    private static final String LOGSTR_VALIDATION_ERRORS = "Errors found while validating exception record id {} - {}";
+    private static final String LOGSTR_VALIDATION_WARNING = "Warnings found while validating exception record id {} - {}";
+
+    private final IdamService idamService;
+    private final CcdService ccdService;
+
+    public SyaAppealValidator(IdamService idamService,
+                              CcdService ccdService,
+                              DwpAddressLookupService dwpAddressLookupService,
+                              AddressValidator addressValidator,
+                              List<String> titles) {
+        super(dwpAddressLookupService, addressValidator, SYA_APPEAL, titles);
+        this.idamService = idamService;
+        this.ccdService = ccdService;
+    }
+
+    public PreSubmitCallbackResponse<SscsCaseData> validateAppeal(CaseDetails<SscsCaseData> caseDetails, Map<String, Object> appealData, boolean ignoreMrnValidation) {
+
+        Map<String, List<String>> errsWarns =
+                validateAppeal(new HashMap<>(), appealData, ignoreMrnValidation, false, false);
+
+        CaseResponse caseValidationResponse = CaseResponse.builder()
+                .errors(errsWarns.get("errors"))
+                .warnings(errsWarns.get("warnings"))
+                .transformedCase(appealData)
+                .build();
+
+        PreSubmitCallbackResponse<SscsCaseData> validationErrorResponse = convertWarningsToErrors(caseDetails.getCaseData(), caseValidationResponse); //convertWarn.. can be in tribs.
```

Comment:
can comment be remove now? Are you moving convertWarningsToErrors into tribs?

Reply:
done